Public/Src/Sandbox/Windows/DetoursServices/DataTypes.h

```diff
@@ -539,6 +539,27 @@ typedef struct ManifestDllBlock_t
 } ManifestDllBlock;
 typedef const ManifestDllBlock * PCManifestDllBlock;
 
+// ==========================================================================
+// == ManifestSubstituteProcessExecutionShim
+// ==========================================================================
+typedef struct ManifestSubstituteProcessExecutionShim_t

```

Comment:
Is this going to be an empty struct when running in Release configuration?  If so, `sizeof(ManifestSubstituteProcessExecutionShim_t)` is technically undefined by the C language standard.  On Windows this seems to always evaluate to 0, in which case everything works just fine; on Mac, we've seen cases where `sizeof()` can be non-zero, in which case things will not work.

I see that all the validation passed for this PR.  Unfortunately though, I don't think we have anything that validates the release configuration on Mac.

Reply:
Didn't know that. I can rearrange to place one of the optional int32 values into the struct itself with a comment that it's needed to anchor the release version of the struct at size > 0

Reply:
I just tried it on my laptop, and I got errors:
```
[0:10] error DX0011: [Pip89B70B5516562FBA, bash, HelloWorld, scriptWithChildProcesses, {}] Process start failed with error code 80131500: [Pip-8523331309700501574 -- Pip89B70B5516562FBA, bash, HelloWorld, scriptWithChildProcesses, {}] Process creation failed: Failed to notify kernel extension about process start, make sure the extension is loaded
```
Please don't merge this yet.  I'll update our CI validation to include some validation for the release bits

Reply:
Your suggestion makes sense.

Reply:
Please pull latest commit with update

Reply:
Works on my laptop!

I just updated CI validation, but feel free to merge without re-running Mac validation

Reply:
Thanks for the extra validation, glad not to break Mac :)

Reply:
Sorry, I was wrong: it only worked because I had all cache hits.

I'm still getting the same errors when it executes processes.

Reply:
I updated CI validations, so if they pass you're good to go.

Reply:
Managed code side of Detours logic was not aligned with the previous commit, new one should be good. Again thanks!